tests/codegen/waiter-tests/src/main/kotlin/com/test/DefaultWaitersTestClient.kt

```diff
@@ -18,75 +18,43 @@ class DefaultWaitersTestClient<T>(resultList: List<Result<T>>) : WaitersTestClie
-        return if (nextResult.isSuccess) (nextResult as Result<GetFunctionLengthResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
-    }
+    override suspend fun getFunctionLength(input: GetFunctionLengthRequest): GetFunctionLengthResponse = findSuccess()
 
-    override suspend fun getFunctionAbs(input: GetFunctionAbsRequest): GetFunctionAbsResponse {
-        val nextResult = results.next()
-        @Suppress("UNCHECKED_CAST")
-        return if (nextResult.isSuccess) (nextResult as Result<GetFunctionAbsResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
-    }
+    override suspend fun getFunctionAbs(input: GetFunctionAbsRequest): GetFunctionAbsResponse = findSuccess()
 
-    override suspend fun getFunctionFloor(input: GetFunctionFloorRequest): GetFunctionFloorResponse {
-        val nextResult = results.next()
-        @Suppress("UNCHECKED_CAST")
-        return if (nextResult.isSuccess) (nextResult as Result<GetFunctionFloorResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
-    }
+    override suspend fun getFunctionFloor(input: GetFunctionFloorRequest): GetFunctionFloorResponse = findSuccess()
 
-    override suspend fun getFunctionCeil(input: GetFunctionCeilRequest): GetFunctionCeilResponse {
-        val nextResult = results.next()
-        @Suppress("UNCHECKED_CAST")
-        return if (nextResult.isSuccess) (nextResult as Result<GetFunctionCeilResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
-    }
+    override suspend fun getFunctionCeil(input: GetFunctionCeilRequest): GetFunctionCeilResponse = findSuccess()
+
+    override suspend fun getSubFieldProjection(input: GetSubFieldProjectionRequest): GetSubFieldProjectionResponse = findSuccess()
+
+    override suspend fun getFunctionSumEquals(input: GetFunctionSumEqualsRequest): GetFunctionSumEqualsResponse = findSuccess()
+
+    override suspend fun getFunctionAvgEquals(input: GetFunctionAvgEqualsRequest): GetFunctionAvgEqualsResponse = findSuccess()
+
+    override suspend fun getFunctionJoinEquals(input: GetFunctionJoinEqualsRequest): GetFunctionJoinEqualsResponse = findSuccess()
+
+    override suspend fun getFunctionStartsWithEquals(input: GetFunctionStartsWithEqualsRequest): GetFunctionStartsWithEqualsResponse = findSuccess()
+
+    override suspend fun getFunctionEndsWithEquals(input: GetFunctionEndsWithEqualsRequest): GetFunctionEndsWithEqualsResponse = findSuccess()
 
-    override suspend fun getSubFieldProjection(input: GetSubFieldProjectionRequest): GetSubFieldProjectionResponse {
+    private fun <Response> findSuccess(): Response {
         val nextResult = results.next()
         @Suppress("UNCHECKED_CAST")
-        return if (nextResult.isSuccess) (nextResult as Result<GetSubFieldProjectionResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
+        return if (nextResult.isSuccess) (nextResult as Result<Response>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
     }
```

Comment:
**Comment**: These look great now!

Also, `findSuccess` is good advice for us all! 😆